app/Http/Middleware/TaskTypeRedirect.php

```diff
@@ -44,51 +44,58 @@ public function handle(Request $request, Closure $next)
-                    break;
-                case 'analysis':
-                    return $this->redirector->route('analysis-producing-choice');
-                    break;
-                case 'progress':
-                    return $this->redirector->route('progress-producing');
-                    break;
-                case 'period':
-                    return $this->redirector->route('period-producing');
-                    break;
-                case 'period-edit':
-                    return $this->redirector->route('period-producing-edit', ['id' => $route->parameter('id')]);
-                    break;
+        
+        if ($student->isStudent()) {
+            if ($student->educationProgram->educationprogramType->isActing()) {
+                switch ($route->getName()) {
+                    case 'home':
+                    case 'default':
+                        return $this->redirector->route('home-acting');
+                        break;
+                    case 'process':
+                        return $this->redirector->route('process-acting');
+                        break;
+                    case 'analysis':
+                        return $this->redirector->route('analysis-acting-choice');
+                        break;
+                    case 'progress':
+                        return $this->redirector->route('progress-acting');
+                        break;
+                    case 'period':
+                        return $this->redirector->route('period-acting');
+                        break;
+                    case 'period-edit':
+                        return $this->redirector->route('period-acting-edit', ['id' => $route->parameter('id')]);
+                        break;
+                }
+            } else {
+                // Assume the user follows an EP of type 'Producing'
+                switch ($route->getName()) {
+                    case 'home':
+                    case 'default':
+                        return $this->redirector->route('home-producing');
+                        break;
+                    case 'process':
+                        return $this->redirector->route('process-producing');
+                        break;
+                    case 'analysis':
+                        return $this->redirector->route('analysis-producing-choice');
+                        break;
+                    case 'progress':
+                        return $this->redirector->route('progress-producing');
+                        break;
+                    case 'period':
+                        return $this->redirector->route('period-producing');
+                        break;
+                    case 'period-edit':
+                        return $this->redirector->route('period-producing-edit', ['id' => $route->parameter('id')]);
+                        break;
+                }
             }
+        } elseif ($student->isAdmin()) {
```

Comment:
Does this mean that users with the admin or teacher role cannot use the Student functionality anymore? As in, they will be forcefully redirected to their respective home page. Maybe add a check in the Admin statement that checks whether the route called is actually for home.

Reply:
Yes that was what I thought. Or do we want them to still use the Student functionality, but only show them a different homepage than the Student homepage?

Reply:
I think for teachers this is a good idea, they don't need the student functionality. However, for admins it is probably best to still allow it, otherwise we will need to keep switching accounts 😅 when developing/testing. Also I think only me and Esther are admins on the live server, so no issues expected over there.